tests/framework/extensions/rke1/nodepools/nodepools.go

```diff
@@ -57,61 +83,83 @@ func NodePoolSetup(client *rancher.Client, nodeRoles []NodeRoles, ClusterID, Nod
 	return &nodePoolConfig, nil
 }
 
-// ScaleWorkerNodePool is a helper method that will add a worker node pool to the existing RKE1 cluster. Once done, it will scale
-// the worker node pool to add a worker node, scale it back down to remove the worker node, and then delete the worker node pool.
-func ScaleWorkerNodePool(client *rancher.Client, nodeRoles []NodeRoles, ClusterID, NodeTemplateID string) error {
-	nodePoolConfig := management.NodePool{
-		ClusterID:               ClusterID,
-		ControlPlane:            false,
-		DeleteNotReadyAfterSecs: 0,
-		Etcd:                    false,
-		HostnamePrefix:          "auto-rke1-scale-" + ClusterID,
-		NodeTemplateID:          NodeTemplateID,
-		Quantity:                1,
-		Worker:                  true,
-	}
-
-	logrus.Infof("Creating new worker node pool...")
-	nodePool, err := client.Management.NodePool.Create(&nodePoolConfig)
+// AllNodeList is a helper method that will return each node in the cluster.
+func AllNodeList(client *rancher.Client, cluster *management.Cluster, nodeRoles NodeRoles) (*management.Node, error) {
+	nodes, err := client.Management.Node.ListAll(&types.ListOpts{
+		Filters: map[string]interface{}{
+			"clusterId": cluster.ID,
+		},
+	})
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	if nodestat.AllManagementNodeReady(client, ClusterID) != nil {
-		return err
+	for _, node := range nodes.Data {
+		if nodeRoles.ControlPlane != node.ControlPlane {
+			continue
+		}
+		if nodeRoles.Etcd != node.Etcd {
+			continue
+		}
+		if nodeRoles.Worker != node.Worker {
+			continue
+		}
+
+		return &node, nil
 	}
 
-	logrus.Infof("New node pool is ready!")
-	nodePoolConfig.Quantity = 2
+	return nil, nil
+}
 
-	logrus.Infof("Scaling node pool to 2 worker nodes...")
-	updatedPool, err := client.Management.NodePool.Update(nodePool, &nodePoolConfig)
+// updateNodePoolQuantity is a helper method that will update the node pool with the desired quantity.
+func updateNodePoolQuantity(client *rancher.Client, cluster *management.Cluster, node *management.Node, nodeRoles NodeRoles) (*management.NodePool, error) {
+	updatedNodePool, err := client.Management.NodePool.ByID(node.NodePoolID)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	if nodestat.AllManagementNodeReady(client, ClusterID) != nil {
-		return err
+	nodePoolConfig := MatchRoleToConfig(nodeRoles, []management.Node{*node})
+	updatedNodePool.Quantity = nodePoolConfig + nodeRoles.Quantity
```

Comment:
I think you should move the log here instead of logging in the test function helper.

Reply:
Are you referring to the `Scaling the node pool` log? I didn't put it here since I wanted to show when the test is scaling up versus scaling down. We could always change the language if it's needed.